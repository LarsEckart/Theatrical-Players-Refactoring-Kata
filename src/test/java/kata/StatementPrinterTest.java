package kata;

import com.github.larseckart.tcr.TestCommitRevertMainExtension;
import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.reporters.Junit5Reporter;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.util.List;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@ExtendWith(TestCommitRevertMainExtension.class)
class StatementPrinterTest {

    @Test
    void golden_master() {
        var performances = List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("hamlet", 29),
                new Performance("hamlet", 31),
                new Performance("hamlet", 15),
                new Performance("as-like", 35),
                new Performance("as-like", 12),
                new Performance("as-like", 20),
                new Performance("othello", 40));
        var plays = Map.of(
                "hamlet", new Play("Hamlet", "tragedy"),
                "as-like", new Play("As You Like It", "comedy"),
                "othello", new Play("Othello", "tragedy"));
        Invoice invoice = new Invoice("BigCo", performances);

        String statement = new StatementPrinter(plays).print(invoice);

        Approvals.verify(statement, new Options(new Junit5Reporter()));
    }

    @Test
    void golden_master_html() {
        var performances = List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("hamlet", 29),
                new Performance("hamlet", 31),
                new Performance("hamlet", 15),
                new Performance("as-like", 35),
                new Performance("as-like", 12),
                new Performance("as-like", 20),
                new Performance("othello", 40));
        var plays = Map.of(
                "hamlet", new Play("Hamlet", "tragedy"),
                "as-like", new Play("As You Like It", "comedy"),
                "othello", new Play("Othello", "tragedy"));
        Invoice invoice = new Invoice("BigCo", performances);

        String statement = new HtmlStatementPrinter(plays).print(invoice);

        Approvals.verifyXml(statement, new Options(new Junit5Reporter()));
    }
}
