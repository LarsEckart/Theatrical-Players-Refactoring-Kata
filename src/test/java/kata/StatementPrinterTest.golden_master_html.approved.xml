<html>
    <h1>Statement for BigCo</h1>
    <table>
        
        <tr>
            <th>play</th><th>seats</th><th>cost</th>
        </tr>
        <tr>
            <td>Hamlet</td><td>55</td><td>$650.00</td>
        </tr>
        
        <tr>
            <td>Hamlet</td><td>30</td><td>$400.00</td>
        </tr>
        
        <tr>
            <td>Hamlet</td><td>29</td><td>$400.00</td>
        </tr>
        
        <tr>
            <td>Hamlet</td><td>31</td><td>$410.00</td>
        </tr>
        
        <tr>
            <td>Hamlet</td><td>15</td><td>$400.00</td>
        </tr>
        
        <tr>
            <td>As You Like It</td><td>35</td><td>$580.00</td>
        </tr>
        
        <tr>
            <td>As You Like It</td><td>12</td><td>$336.00</td>
        </tr>
        
        <tr>
            <td>As You Like It</td><td>20</td><td>$360.00</td>
        </tr>
        
        <tr>
            <td>Othello</td><td>40</td><td>$500.00</td>
        </tr>
        
    </table>
    
    <p>
        Amount owed is <em>$4,036.00</em>
    </p>
    
    <p>
        You earned <em>54</em> credits
    </p>
    
</html>
